locksmith/src/operations/checkoutConfigOperations.ts

```diff
@@ -95,6 +94,17 @@ export const saveCheckoutConfig = async ({
     checkoutConfigData.referrer = createdBy
   }
 
+  // Check if the user is authorized to save/update the config
+  if (id) {
+    const existingConfig = await CheckoutConfig.findOne({ where: { id } })
+    if (
+      existingConfig &&
+      !(await isUserAuthorized(createdBy, existingConfig))
```

Comment:
for clarity, please let's rename `createdBy` to something like `user` because the user could be just a lock manager, not the creator of the config.